src/ParameterAttributeCollector.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace olvlvl\ComposerAttributeCollector;
+
+
+use Composer\IO\IOInterface;
+use PhpParser\Parser;
+
+/**
+ * @internal
+ */
+class ParameterAttributeCollector
+{
+    private IOInterface $io;
+    private Parser $parser;
+
+    public function __construct(IOInterface $io, Parser $parser)
+    {
+        $this->io = $io;
+        $this->parser = $parser;
+    }
+
+    /**
+     * @return array<TransientTargetMethodParameter>
+     */
+    public function collectAttributes(\ReflectionFunctionAbstract $reflectionFunctionAbstract): array
```

Comment:
I have build this collector on `ReflectionFunctionAbstract`, so we can easily add support for function parameters